### PR TITLE
[ALLUXIO-1863][ALLUXIO-1729] Deflake tests with time based periodical heartbeats

### PR DESCRIPTION
**Pull request by uronce-cc**
https://alluxio.atlassian.net/browse/ALLUXIO-1863
https://alluxio.atlassian.net/browse/ALLUXIO-1729

### CODE REVIEW

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

File: tests/src/test/java/alluxio/client/IsolatedFileSystemIntegrationTest.java [hunk 1]
```diff
@@ -110,13 +118,17 @@ public void lockBlockTest2() throws IOException, AlluxioException {
       Assert.assertTrue(info.getInMemoryPercentage() == 100);
     }
     // Sleep to ensure eviction has been reported to master
```
Comment (jsimsa) [inline tests/src/test/java/alluxio/client/IsolatedFileSystemIntegrationTest.java]: remove or update outdated comment

**Comment by jsimsa** [comment]
@uronce-cc thank you for deflaking all of these tests! ... one minor comment, otherwise LGTM

File: tests/src/test/java/alluxio/client/IsolatedFileSystemIntegrationTest.java (hunk 1 above)
Comment (uronce-cc) [inline tests/src/test/java/alluxio/client/IsolatedFileSystemIntegrationTest.java]: oh, missed that

**Comment by jsimsa** [comment]
LGTM

**Comment by yupeng9** [comment]
Thanks for the fix! Could you also update the title?

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
@yupeng9 Title updated, new ticket created.

**Comment by yupeng9** [comment]
LGTM.

**Comment by yupeng9** [comment]
Could you sign the license agreement?

**Comment by uronce-cc** [comment]
@yupeng9 CLA signed.

**Comment by uronce-cc** [comment]
I greped "sleep" in tests/src/java, and replaced those that may cause flaky tests due to time based heartbeat with scheduled heartbeat.

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**